### DISCUSSION TITLE
How to know how many consumers are in a group

### DISCUSSION THREAD

**Discussion by AzikDeveloper** [django/channels#1837] [General]
I am new to django channels and I am struggling to count the number of consumers in a group. All the answers for this question is old so I am asking it here.

**Comment by yandiro** [comment]
Apparently this is a long-living question. I read some feature requests from 2016 and what they suggest is for us to keep track of that ourselves, but don't really give an example how.

I think it would be something like increment something on connection and decrement on disconnection. This is what I am doing, actually, but we want to rethink that. So I am looking for a better way to do so.

**Comment by carltongibson** [comment]
Have you seen https://django-channels-presence.readthedocs.io/en/latest/ ?

**Reply by yandiro** [reply]
I actually have, but this package's last update was 2 years ago and I don't think it is compatible with Channels v3. Have you used it @carltongibson?

**Reply by carltongibson** [reply]
I'm not using it myself, but others are. Looking at the code, there's nothing ASGI v3 related — it doesn't provide ASGI callables — so no reason that it'd need major changes for Channels v3. 

I mentioned it though as it targets the exact question here, so it's likely what's required.
